lib/BackbeatConsumer.js

```diff
@@ -99,6 +105,30 @@ class BackbeatConsumer extends EventEmitter {
                         offset,
                     });
                     this.emit('error', err, entry);
+                } else {
+                    const qEntry = QueueEntry.createFromKafkaEntry(entry);
+                    if (typeof qEntry.getReducedLocations === 'function') {
+                        const locations = qEntry.getReducedLocations();
+                        const bytes = locations.reduce((sum, item) =>
+                            sum + item.size, 0);
+
+                        if (!this._metricsStore[qEntry.bucket]) {
```

Comment:
Same thing about putting the metrics update code in a separate method. Actually you could have this code in a new `QueueEntry.updateMetrics(metricsStore)` method for example, that would avoid duplicating the logic, since all needed knowledge is already in the queue entry. Ideally `metricsStore` could be managed by a separate class to embed the logic of counting operations per bucket in one code unit.

Reply:
Great idea :)